src/Methods/HigherOrderTapProxyExtension.php

```diff
@@ -35,9 +36,13 @@ public function getMethod(
         ClassReflection $classReflection,
         string $methodName
     ): MethodReflection {
-        /** @var ObjectType $templateType */
+        /** @var ObjectType|ThisType $templateType */
         $templateType = $classReflection->getActiveTemplateTypeMap()->getType('TClass');
 
+        if ($templateType instanceof ThisType) {
+            $templateType = $templateType->getStaticObjectType();
```

Comment:
@canvural Honestly, I'm not really sure whether this is the correct way to do it. This works, but it doesn't really feel like the *right* way to do it. I'm still too unfamiliar with the PHPStan codebase. Maybe there's a better way?

Reply:
Maybe @ondrejmirtes can help here.

Reply:
I don't have the context - can you show me what you're trying to achieve on an analysed code sample?

Reply:
I realised I could move this logic to tap's DynamicFunctionReturnTypeExtension instead, which makes a lot more sense, and makes it a bit simpler.

So I suppose this can be resolved.